### PR TITLE
Linting

### PR DESCRIPTION
**Pull request by mloning**
#### Reference Issues/PRs
#50 

#### What does this implement/fix? Explain your changes.
Cleans up code and enforces PEP8 on Azure Pipelines using flake8 package

#### Any other comments?
* potential problem in MCNN classifer for `n_train_batch` around line 219 in if-statement (now commented out), variable unused, wrong name?
* runs all tests, including "slow" ones
* removed long comment in MCNN classifier, move to relevant issue if important
* `__all__` is required in `__init__` files for linting to pass, otherwise unused variable error is raised
* excludes `experimental/` from linting for now
* I recommend to set IDE max line length to PEP8 standard of 79 characters (had to go through quite a few lines manually because pycharm couldn't clean them up automatically)

### CODE REVIEW

**Comment by mloning** [comment]
Error summary: 
* on latest Linux with TF 1.15 MLP accuracy test fails 
* on TF 1.9 tests time out (job limit is 60 min)

Any suggestions?

**Comment by James-Large** [comment]
MLP accuracy test is almost certainly just annoying chance, it is only just below the bound
[test](https://github.com/sktime/sktime-dl/blob/dev/sktime_dl/deeplearning/tests/test_accuracy.py#L85), [result: assert (0.9465500485908649 > 0.948)](https://github.com/sktime/sktime-dl/pull/51/checks?check_run_id=640524284)

Problem is with reproducibility in deep learning land. We can try to go to extremes to force seeds on everything, though it may not be possible to make everything truly reproducible on distributed/varying CI hardware and OS environments ([1](https://stackoverflow.com/a/52897289), [2](https://keras.io/getting-started/faq/#how-can-i-obtain-reproducible-results-using-keras-during-development)), or we basically keep these tests as they are in concept (achieve acceptably-close-to-published accuracy) and either
1) Widen the acceptable range, i.e. put [ACCURACY_DEVIATION_THRESHOLD up to 5](https://github.com/sktime/sktime-dl/blob/dev/sktime_dl/deeplearning/tests/test_accuracy.py#L29). Just have to accept the test failure once every 1000+ times
2) Add another condition to the rerun_filter (is_not_value_error) at the above link again, where if it's still kinda close, but outside the range, rerun the test. Still accept the failure once every 1000+ times.. 



On the time out, was the problem 60 mins total or no output for 10 mins? Assuming the Travis checks ran for 35 min total https://travis-ci.com/github/sktime/sktime-dl/jobs/326990113#L1192

This is just symptomatic of running deep learners on cpu, presumably single threaded, and probably not with a high clock rate one. Still funny it takes >10mins to go through ItalyPowerDemand though. There are mechanisms like [travis wait](https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received) it seems, but that's probably a bad road to start going down, ideally they wouldn't take this long in the first place. Otherwise, if _any_ output would do, we could up the verbosity during tests, means that the 10 minute timer is not reached (if indeed I'm looking at the right thing and it's not a 60 min global timeout).



The chance of randomness causing a failure, plus the super slow run times maybe means these tests need to be revisited, or just disabled as CI tests, or _most_ of them disabled. There's still the general 'runnable' tests that use tiny inputs to test that outputs are produced, but having a mechanism to ensure continued accuracy would be good

**Comment by James-Large** [comment]
On the MCNN, think I know what's happened, that'll have been there since back when Aaron ported it over at the sprint week, mistake when chopping up MCNN's tuning procedure into neat fit/predict stages, hopefully fixing wont break the whole thing...

**Comment by James-Large** [comment]
Ah I see, on the re-run travis failed with 10 mins no output, but Azure with 60 mins total.

**Comment by mloning** [comment]
I believe this is a bug, `test_fcn_accuracy` timed out after 6 hours on Azure on Windows with Python 3.6 and TF 1.9. I'm adding pytest markers to conditionally skip tests depending on versions/platforms we use.

**Comment by Withington** [comment]
Thanks for making all the PEP8 changes @mloning

**Comment by Withington** [comment]
Slow running tests - it looks like Azure can be set up to run [scheduled jobs](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/scheduled-triggers?view=azure-devops), e.g. only run the slow tests once a day, overnight.
 
I agree that its unlikely that we'll ever get reproducible numbers. 

I like the idea of creating tiny datasets to train accurate but quick models. However, in the case of sktime-dl, we do need to demonstrate that the models that we have reproduced from the literature do produce the same results as were published. We could easily drift away from this unless we continue to test for it. For the same reason, I'd rather keep  ACCURACY_DEVIATION_THRESHOLD at 3. 

TF2 did introduce speed ups; another reason for us to drop support/testing for TF1.9.

Anyway, if we could run the slow tests only once a day, I think we'd be ok.

**Comment by mloning** [comment]
Okay, I've added the option to turn on/off slow tests via the env variable `SLOW_TEST`, default is false, and now back to running only the faster tests.

I suggest to open a new PR for the scheduled jobs, my idea was to overwrite `SLOW_TEST` for the scheduled tests, but we still need to mark all the tests that fail on the different platforms/versions if we want to do that, or we only include Linux, that may be enough for guaranteeing that we can reproduce published results.

**Comment by mloning** [comment]
Okay, this is ready to go from my side. Please review/merge! 🚀
